Tests/Algorithm/Framework/Portfolio/MeanReversionPortfolioConstructionModelTest.cs

```diff
@@ -95,28 +142,97 @@ public virtual void SetUp()
+            var spy = _algorithm.AddEquity("SPY");
+
+            var insights = new List<Insight>
             {
-                var exception = Assert.Throws<ArgumentException>(() => GetPortfolioConstructionModel(language, bias, Resolution.Daily));
-                Assert.That(exception.Message, Is.EqualTo("Long position must be allowed in MeanReversionPortfolioConstructionModel."));
+                new Insight(_nowUtc, aapl.Symbol, TimeSpan.FromDays(1), InsightType.Price, InsightDirection.Up, null, null),
+                new Insight(_nowUtc, spy.Symbol, TimeSpan.FromDays(1), InsightType.Price, InsightDirection.Up, null, null),
+            };
+            _algorithm.PortfolioConstruction.OnSecuritiesChanged(_algorithm, SecurityChangesTests.AddedNonInternal(aapl, spy));
+            
+            var history = _algorithm.History<TradeBar>(new[] {aapl.Symbol, spy.Symbol}, 2, Resolution.Daily);
+            var aaplHist = history.Select(slice => slice[aapl.Symbol].Close);
+            var spyHist = history.Select(slice => slice[spy.Symbol].Close);
+            var aaplRelative = (double) (aaplHist.Last() / aaplHist.Average());
+            var spyRelative = (double) (spyHist.Last() / spyHist.Average());
+
+            var result = model.TestGetPriceRelatives(insights).Select(x => Math.Round(x, 8)).ToArray();
+            var expected = new double[] {aaplRelative, spyRelative};
+            expected = expected.Select(x => Math.Round(x, 8)).ToArray();
+            Assert.AreEqual(expected, result);
+        }
+
+        [Test]
+        public void GetPriceRelativesWithInsightMagnitude()
+        {
+            var model = new TestMeanReversionPortfolioConstructionModel();
+            SetPortfolioConstruction(Language.CSharp, PortfolioBias.Long, model);
+
+            var aapl = _algorithm.AddEquity("AAPL");
+            var spy = _algorithm.AddEquity("SPY");
+
+            var insights = new List<Insight>
+            {
+                new Insight(_nowUtc, aapl.Symbol, TimeSpan.FromDays(1), InsightType.Price, InsightDirection.Up, 1, null),
+                new Insight(_nowUtc, spy.Symbol, TimeSpan.FromDays(1), InsightType.Price, InsightDirection.Up, -0.5, null),
+            };
+            _algorithm.PortfolioConstruction.OnSecuritiesChanged(_algorithm, SecurityChangesTests.AddedNonInternal(aapl, spy));
+            
+            var result = model.TestGetPriceRelatives(insights);
+            var expected = new double[] {2d, 0.5d};
+            Assert.AreEqual(expected, result);
+        }
+
+        [TestCase(Language.CSharp, 1)]
+        [TestCase(Language.Python, 1)]
+        [TestCase(Language.CSharp, 0.5)]
+        [TestCase(Language.Python, 0.5)]
+        [TestCase(Language.CSharp, 0)]
+        [TestCase(Language.Python, 0)]
+        public void SimplexProjection(Language language, double regulator)
+        {
+            var model = GetPortfolioConstructionModel(language, PortfolioBias.Long, Resolution.Daily);
```

Comment:
`model` isn't used really, could call the python version for python language too I believe. Same for `GetPriceRelativesWithInsightMagnitude` & `GetPriceRelatives`